iothub/device/src/Transport/AmqpIoT/AmqpIoTConnector.cs

```diff
@@ -62,81 +57,46 @@ public async Task<AmqpIoTConnection> OpenConnectionAsync(TimeSpan timeout)
                 amqpConnectionSettings.IdleTimeOut = Convert.ToUInt32(idleTimeout.TotalMilliseconds);
             }
 
-            var amqpIoTTransport = new AmqpIoTTransport(amqpSettings, _amqpTransportSettings, _hostName, s_disableServerCertificateValidation);
+            _amqpIotTransport = new AmqpIoTTransport(amqpSettings, _amqpTransportSettings, _hostName, s_disableServerCertificateValidation);
 
-            TransportBase transportBase = await amqpIoTTransport.InitializeAsync(timeout).ConfigureAwait(false);
+            TransportBase transportBase = await _amqpIotTransport.InitializeAsync(timeout).ConfigureAwait(false);
             try
             {
                 var amqpConnection = new AmqpConnection(transportBase, amqpSettings, amqpConnectionSettings);
-                AmqpIoTConnection amqpIoTConnection = new AmqpIoTConnection(amqpConnection);
+                var amqpIoTConnection = new AmqpIoTConnection(amqpConnection);
                 amqpConnection.Closed += amqpIoTConnection.AmqpConnectionClosed;
                 await amqpConnection.OpenAsync(timeout).ConfigureAwait(false);
 
-                if (Logging.IsEnabled)
-                {
-                    Logging.Exit(this, timeout, $"{nameof(OpenConnectionAsync)}");
-                }
+                Logging.Exit(this, timeout, $"{nameof(OpenConnectionAsync)}");
 
                 return amqpIoTConnection;
             }
-            catch (Exception e) when (!e.IsFatal())
+            catch (Exception ex) when (!ex.IsFatal())
             {
                 transportBase?.Close();
+                _amqpIotTransport?.Dispose();
```

Comment:
_amqpIotTransport = null ?

Reply:
Why?

Reply:
I'll fix this in my next PR. I understand now.